brozzler/js-templates/umbraBehavior.js.j2

```diff
@@ -0,0 +1,174 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var umbraBehavior = {
+    IDLE_TIMEOUT_SEC : 10,
+
+    alreadyDone : [],
+    idleSince : null,
+    intervalId : null,
+    actions : {{actions|json}},
+
+    intervalFunc: function() {
+        var didSomething = false;
+        var somethingLeftBelow = false;
+        var somethingLeftAbove = false;
+        var actionsLength = this.actions.length;
+
+        for (var k = 0; k < actionsLength; k++) {
+            var selector = this.actions[k].selector;
+            var action = this.actions[k].do ? this.actions[k].do : 'click';
+            var limit = this.actions[k].limit ? this.actions[k].limit : 0;
+            if (limit && this.actions[k].alreadyDone && this.actions[k].alreadyDone.length >= limit) {
+                continue;
+            }
+            if (limit && !(this.actions[k].alreadyDone)) {
+                this.actions[k].alreadyDone = [];
+            }
+
+            var iframes = document.querySelectorAll("iframe");
+            var documents = Array(iframes.length + 1);
+            documents[0] = document;
+
+            iframesLength = iframes.length;
+            for (var i = 0; i < iframesLength; i++) {
+                documents[i+1] = iframes[i].contentWindow.document;
+            }
+
+            documentsLength = documents.length;
+            for (var j = 0; j < documentsLength; j++) {
+
+                var doTargets = documents[j].querySelectorAll(selector);
+                if (doTargets == []) {
+                    continue;
+                }
+
+                doTargetsLength = doTargets.length;
+                for ( var i = 0; i < doTargetsLength; i++) {
+                    if (limit && this.actions[k].alreadyDone && this.actions[k].alreadyDone.length >= limit) {
+                        break;
+                    }
+                    if (this.alreadyDone.indexOf(doTargets[i]) > -1) {
+                        continue;
+                    }
+                    if (!this.isVisible(doTargets[i])) {
+                        continue;
+                    }
+
+                    var where = this.aboveBelowOrOnScreen(doTargets[i]);
+                    if (where == 0) {
+                        this.doTarget(doTargets[i], action);
+                        didSomething = true;
+                        if (this.actions[k].alreadyDone) {
+                            this.actions[k].alreadyDone.push(doTargets[i]);
+                        }
+                    } else if (where > 0) {
+                        somethingLeftBelow = true;
+                    } else if (where < 0) {
+                        somethingLeftAbove = true;
+                    }
+                    if (didSomething) {
+                        break; // break from doTargets loop, but not from documents loop
+                    }
+                }
+            }
+
```

Comment:
👇 should this section be outside of the `for (var k = 0; k < actionsLength; k++)` loop?